rust/Earthfile

```diff
@@ -45,61 +53,58 @@ CARGO:
     DO +CHECK_INITED
     ARG --required args
     ARG output
-    ARG TMP_FOLDER="/tmp/earthly/lib/rust"
     IF [ "$EARTHLY_KEEP_FINGERPRINTS" = "false" ]
        DO +REMOVE_SOURCE_FINGERPRINTS
     END
-    DO +RUN_WITH_CACHE --command="set -e;
-        echo \"Running cargo $args\" ;
-        cargo $args;
-        if [ -n \"$output\" ]; then
-          echo \"Copying output files\" ;
-          mkdir -p $TMP_FOLDER;
-          cd target;
-          find . -type f -regextype posix-egrep -regex \"./$output\" -exec cp --parents \{\} $TMP_FOLDER \; ;
-          cd ..;
-        fi;
-        echo \"Running cargo sweep -r -t $EARTHLY_SWEEP_DAYS\" ;
-        cargo sweep -r -t $EARTHLY_SWEEP_DAYS;
-        echo \"Running cargo sweep -r -i\" ;
-        cargo sweep -r -i;"
+    DO +GET_RUST_CACHE_MOUNTS
+    RUN --mount=$EARTHLY_RUST_CARGO_HOME_CACHE --mount=$EARTHLY_RUST_TARGET_CACHE \
+      set -e; \
+      cargo $args; \
+      cargo sweep -r -t $EARTHLY_SWEEP_DAYS; \
+      cargo sweep -r -i;
     IF [ "$output" != "" ]
-      RUN mkdir -p target; \
-          mv $TMP_FOLDER/* target 2>/dev/null || echo "no files found within ./target matching the provided output regexp" ;
+      DO +COPY_OUTPUT --output=$output
     END
 
-# RUN_WITH_CACHE runs the passed command with the CARGO caches mounted.
-# Notice that in order to run this function, +INIT must be called first. This function exports the target cache mount ID under the env entry: $TARGET_CACHE_ID.
-# Arguments:
-#   - command (required): Command to run, can be any expression.
-#   - EARTHLY_CARGO_HOME_CACHE_ID: ID of the cargo home cache mount. By default: $EARTHLY_CARGO_HOME_CACHE_ID as exported by +INIT
-#   - target_cache_id: ID of the target cache mount. By default: ${EARTHLY_CARGO_HOME_CACHE_ID}#${EARTHLY_TARGET_NAME}
-#
-RUN_WITH_CACHE:
```

Comment:
This is still mentioned in the README, should be updated to talk about `GET_RUST_CACHE_MOUNTS` instead.